src/context/index.ts

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { computed, IArrayWillSplice, intercept, observable, observe } from 'mobx'
+import { Log, Result } from 'sarif'
+import { commands, DiagnosticSeverity, ExtensionContext, languages, Memento, Range, Selection, TextDocument, ThemeColor, Uri, window, workspace } from 'vscode'
+import { mapDistinct, _Region, parseRegion } from '../shared'
+import '../shared/extension'
+import { Baser } from './Baser'
+import { loadLogs } from './loadLogs'
+import { Panel } from './Panel'
+
+declare module 'vscode' {
+	interface Diagnostic {
+		_result: Result
+	}
+}
+
+export const regionToSelection = (doc: TextDocument, region: _Region | undefined) => {
+	if (!region) return new Selection(0, 0, 0, 0) // TODO: Decide if empty regions should be pre-filtered.
+	return Array.isArray(region)
+		? (region.length === 4
+			? new Selection(...region)
+			: (() => {
+				const [byteOffset, byteLength] = region
+				const startColRaw = byteOffset % 16
+				const endColRaw = (byteOffset + byteLength) % 16
+				return new Selection(
+					Math.floor(byteOffset / 16),
+					10 + startColRaw + Math.floor(startColRaw / 2),
+					Math.floor((byteOffset + byteLength) / 16),
+					10 + endColRaw + Math.floor(endColRaw / 2),
+				)
+			})())
+		: (() => {
+			const line = doc.lineAt(region)
+			return new Selection(
+				line.range.start.line,
+				line.firstNonWhitespaceCharacterIndex,
+				line.range.end.line,
+				line.range.end.character,
+			)
+		})()
```

Comment:
I believe if statements would be much more readable here. This ternary operator spans ~20 lines.

Reply:
Agreed. This one grew. Will fix.

Reply:
Fixed.